src/build.py

```diff
@@ -488,18 +498,32 @@ def SyncPrebuiltCMake(name, src_dir, git_repo):
       raise
 
 
+def SyncTarball(out_dir, name, version, url, tar):
+  if os.path.isdir(out_dir):
+    print '%s directory already exists' % name
+    return
+  print 'Downloading %s %s from %s' % (name, version, url)
+  f = urllib2.urlopen(url)
+  print 'URL: %s' % f.geturl()
+  print 'Info: %s' % f.info()
+  with open(tar, 'wb') as out:
+    out.write(f.read())
+  proc.check_call(['tar', '-xvf', tar], cwd=WORK_DIR)
+  assert os.path.isdir(out_dir), 'Untar should produce %s' % out_dir
+
+
 def SyncOCaml(name, src_dir, git_repo):
-  if os.path.isdir(src_dir):
-    print 'OCaml directory already exists'
-  else:
-    print 'Downloading OCaml %s from %s' % (OCAML_VERSION, OCAML_URL)
-    f = urllib2.urlopen(OCAML_URL)
-    print 'URL: %s' % f.geturl()
-    print 'Info: %s' % f.info()
-    with open(OCAML_TAR, 'wb') as out:
-      out.write(f.read())
-    proc.check_call(['tar', '-xvf', OCAML_TAR], cwd=WORK_DIR)
-    assert os.path.isdir(OCAML_DIR), 'Untar should produce %s' % OCAML_DIR
+  return SyncTarball(src_dir, 'OCaml', OCAML_VERSION, OCAML_URL, OCAML_TAR)
+
+
+def SyncPrebuiltNodeJS(name, src_dir, git_repo):
+  extension = {'darwin': 'gz', 'linux2': 'xz'}[sys.platform]
```

Comment:
Should we make a struct with the platform specific junk in it? So here we could call `PlatformJunk().zipExtension` and elsewhere have

```python
RAW_JUNK = {
  'darwin': {
    zipExtension: 'gz',
    nodePlatformName: 'darwin-x64',
# ...
}
def PlatformJunk():
  return RAW_JUNK[sys.platform]
```